tracee-rules/input.go

```diff
@@ -66,8 +68,132 @@ func setupTraceeSource(traceeFilePath string) (chan types.Event, error) {
 				res <- event
 			}
 		}
-		f.Close()
+		opts.inputFile.Close()
+		close(res)
+	}()
+	return res, nil
+}
+
+func setupJSONInputSource(opts *traceeInputOptions) (chan types.Event, error) {
+	res := make(chan types.Event)
+	scanner := bufio.NewScanner(opts.inputFile)
+	go func() {
+		for scanner.Scan() {
+			event := scanner.Bytes()
+			var e tracee.Event
+			err := json.Unmarshal(event, &e)
+			if err != nil {
+				log.Printf("invalid json in %s: %v", string(event), err)
+			}
+			res <- tracee.Event(e)
+		}
+		opts.inputFile.Close()
 		close(res)
 	}()
 	return res, nil
 }
+
+func parseTraceeInputOptions(inputOptions []string) (*traceeInputOptions, error) {
```

Comment:
Same question as above - is there something specific to tracee here?
If yes, is it possible to make this function generic?

Reply:
_(see comment above)_